app/src/main/java/com/benefit/ui/products/ProductsPageUI.java

```diff
@@ -0,0 +1,252 @@
+        } else {
+            writeFilter(currentCategory.getName(), currentFilterLayout);
+        }
+    }
+
+    private void writeOrSign(LinearLayout currentFilterLayout) {
+        TextView orSign = new TextView(view.getContext());
+        orSign.setText(view.getResources().getString(R.string.divider));
+        currentFilterLayout.addView(orSign);
+    }
+
+    private void writeFilter(String name, LinearLayout currentFilterLayout) {
+        TextView filterText = new TextView(view.getContext());
+        filterText.setText(name);
+        setTextLayoutMargins(filterText);
+        currentFilterLayout.addView(filterText);
+    }
+
+    private void setTextLayoutMargins(TextView filterText) {
+        LinearLayout.LayoutParams layoutParams = new
+                LinearLayout.LayoutParams(LinearLayout.LayoutParams.WRAP_CONTENT,
+                LinearLayout.LayoutParams.WRAP_CONTENT);
+        layoutParams.setMargins(Converters.convertDpToPx(5), 0,
+                Converters.convertDpToPx(5), 0);
+        filterText.setLayoutParams(layoutParams);
+    }
+
+    /**
+     * Undims the background
+     */
+    public void undimBackground(){
+        ViewOverlay overlay = view.getOverlay();
+        overlay.clear();
+    }
+
+    /**
+     * opens the filter popup
+     *
+     * @param currentFilters the current filters wanted
+     */
+    public void openFilter(View view, List<PropertyName> filters, Map<String, List<String>> currentFilters) {
+
+        RelativeLayout viewGroup = view.findViewById(R.id.filter_popup);
+        LayoutInflater layoutInflater = (LayoutInflater) view.getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+        popupView = layoutInflater.inflate(R.layout.filter, viewGroup);
+        placePopupOnScreen();
+        populatePopup(filters, currentFilters);
+        applyDim( 0.8f);
+    }
+
+    private void applyDim(float dimAmount){
+        Drawable dim = new ColorDrawable(Color.BLACK);
+        dim.setBounds(0, 0, view.getWidth(), view.getHeight());
+        dim.setAlpha((int) (255 * dimAmount));
+        ViewOverlay overlay = view.getOverlay();
+        overlay.add(dim);
+    }
+
+    private void populatePopup(List<PropertyName> filters, Map<String, List<String>> currentFilters) {
+        filterPopup = new FilterPopup(popupView, filters);
+        filterPopup.populateFilter(currentFilters);
+    }
+
+    private void placePopupOnScreen() {
+        popup = new PopupWindow(view.getContext());
+        setPopupAttributes(popup, popupView);
+        popup.showAtLocation(popupView, Gravity.END, 0, 0);
+    }
+
+    private void setPopupAttributes(PopupWindow popup, View layout) {
+        int width = (int) ((view.getResources().getDisplayMetrics().widthPixels) / 1.2);
+        int height = view.getResources().getDisplayMetrics().heightPixels;
+        popup.setContentView(layout);
+        popup.setWidth(width);
+        popup.setHeight(height);
+        popup.setFocusable(true);
+    }
+
+    /**
```

Comment:
I find all these comments redundant, what do you say about removing them from this file?

Reply:
I agree. But they asked for for comments on all public methods, no?

Reply:
who's 'they'? With Aviad we agreed on classes documentation